### PR TITLE
WT-12378 Not enough memory error on arm64-small

### PR DESCRIPTION
**Pull request by bzm-jas**
Fixed this error by assigning the tests that requires more memory to arm64-medium.

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -5541,6 +5541,8 @@ buildvariants:
   tasks:
     - name: compile
     - name: make-check-test
+      # Using medium ditro as more memory is required. See WT-12378.
```
Comment (clarissecheah) [inline test/evergreen.yml:5544]: Typo "distro" . Also maybe just keep one occurrence of this comment here, something like:

"# Using medium distro for a few tests here as more memory is required. See WT-12378."

Comment (bzm-jas) [reply test/evergreen.yml]: fixed here: [a6f2964](https://github.com/wiredtiger/wiredtiger/pull/10260/commits/a6f2964685674fa985ae51fae6ee8a8dcee8c102)

File: test/evergreen.yml [hunk 2]
```diff
@@ -5538,9 +5538,11 @@ buildvariants:
     ENABLE_TCMALLOC: -DENABLE_TCMALLOC=1
     num_jobs: $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
     cmake_generator: "Unix Makefiles"
+  # Using large distro for a few tests here as more memory is required. See WT-12378.
```
Comment (etienneptl) [inline test/evergreen.yml:5541]: We try not to mention tickets in the code. One can use `git blame` to check where those changes come from!
```suggestion
  # Using large distro for a few tests here as more memory is required.
```

File: test/evergreen.yml [hunk 3]
```diff
@@ -5775,6 +5779,8 @@ buildvariants:
   tasks:
     - name: compile
     - name: make-check-test
+      # Using large distro as more memory is required. See WT-12378.
```
Comment (etienneptl) [inline test/evergreen.yml:5782]: ditto
```suggestion
      # Using large distro as more memory is required.
```

**Review by etienneptl — APPROVED** [review]
LGTM, 2 small comments. Thank you for all the testing!

**Review by clarissecheah — APPROVED** [review]
LGTM! Nice one

**Pull request merged**